index.js

```diff
@@ -123,12 +131,15 @@ module.exports.init = function () {
     });
 
   cluster.once('listening', function (worker, address) {
-    debug('cluster is listening on %s', address.port || address.fd);
-    if (address.address &&
-        address.address[0] === '/' &&
-        unix_sockets.indexOf(address.address) === -1) {
-      unix_sockets.push(address.address);
-      fs.chmodSync(address.address, '664');
+    debug('cluster is listening on %s', address.port || address.address);
+  }).on('listening', function (worker, address) {
+    if (address.addressType === -1) { // https://nodejs.org/api/cluster.html#cluster_event_listening_1
+      unix_sockets.add(address.address);
+
+      // due to https://github.com/nodejs/node/issues/19729 the underlying path may have
+      // been unlinked after this cluster has started. ensure that the permissions are
+      // correct whenever a new worker has bound to a UNIX socket.
+      ensurePermsAsync(address.address, '664');
```

Comment:
I saw you changed Sync => Async, any reason related to the issue?

Reply:
Only because this is now happening "continually" as new workers come up, so I am avoiding blocking the event loop of the master process for I/O.